common/src/stack/command/stack/commands/set/host/power/imp_ipmi.py

```diff
@@ -0,0 +1,35 @@
+# @copyright@
+# Copyright (c) 2006 - 2019 Teradata
+# All rights reserved. Stacki(r) v5.x stacki.com
+# https://github.com/Teradata/stacki/blob/master/LICENSE.txt
+# @copyright@
+
+import stack.commands
+from collections import namedtuple
+
+class Implementation(stack.commands.Implementation):
+	def run(self, args):
+		host = args[0]
+		cmd = args[1]
+		ipmi_ip = ''
+		impl_output = namedtuple('output', 'out debug success')
```

Comment:
It seems odd to me to specify the same `namedtuple` type in each implementation, since it essentially amounts to duplicated code and requires that any new implementation writers remember to create a `namedtuple` with the exact same interface.

I would recommend passing down the `namedtuple` class to use as an argument to the implementations from the stack command. That way, the interface is fixed by the command class and the `namedtuple` structure is defined in one place.

This is what I had to do in some of the `stack list * firmware` commands. I passed down a "common key", which was a `namedtuple` with certain fields, to the plugins from the command class. This allowed the command code to rely on that "common key" when aggregating plugin results.

Reply:
That sounds good, it also addresses a concern of mine that the implementation was dictating a bit how the information was being returned to the command, since it set the named tuple field names. With this now the command class can control the field names.